sql-plugin/src/main/scala/com/nvidia/spark/rapids/ParquetCachedBatchSerializer.scala

```diff
@@ -1095,13 +1105,11 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
      * relationship. Each ColumnarBatch is converted to a single ParquetCachedBatch when next()
      * is called on this iterator
      */
-    class ColumnarBatchToCachedBatchIterator extends InternalRowToCachedBatchIterator {
-      override def getIterator: Iterator[InternalRow] = {
-
-        new Iterator[InternalRow] {
-          // We have to check for null context because of the unit test
-          Option(TaskContext.get).foreach(_.addTaskCompletionListener[Unit](_ => hostBatch.close()))
```

Comment:
We cannot remove this, because there's no guarantee the iterator is completely drained before the task completes (e.g.: limit query, task exception, etc.).

Reply:
aah. didn't think of an early exit. 

I have added this back.